modules/data.atmosphere/R/tdm_nc2dat.train.R

```diff
@@ -0,0 +1,209 @@
+##' @param lon.in - longitude as numeric
+##' @param upscale - Upscale can either be set for FALSE (leave alone) or to the temporal resolution you want to aggregate to
+#                    options are: "year", "doy" (day of year), or "hour"
+##' @param overwrite
+##' @param verbose
+# -----------------------------------
+#----------------------------------------------------------------------
+# Begin Script
+#----------------------------------------------------------------------
+nc2dat.train <- function(outfolder, in.path, in.prefix, start_date, end_date, 
+    upscale = FALSE, overwrite = FALSE, verbose = FALSE, ...) {
+    
+    start_date <- as.POSIXlt(start_date, tz = "UTC")
+    end_date <- as.POSIXlt(end_date, tz = "UTC")
+    
+    start_year <- lubridate::year(start_date)
+    end_year <- lubridate::year(end_date)
+    
+    yr_seq <- seq(start_year, end_year)
+
+    #------------ Read in the data
+    
+    input_met <- list()
+    for (j in seq_along(yr_seq)) {
+        input_met[[j]] <- file.path(in.path, paste0(in.prefix, ".", yr_seq[j], 
+            ".nc", sep = ""))
+    }
+    
+    
+    vars.info <- data.frame(CF.name = c("air_temperature", "air_temperature_max", 
+        "air_temperature_min", "surface_downwelling_longwave_flux_in_air", 
+        "air_pressure", "surface_downwelling_shortwave_flux_in_air", "eastward_wind", 
+        "northward_wind", "specific_humidity", "precipitation_flux"), units = c("Kelvin", 
+        "Kelvin", "Kelvin", "W/m2", "Pascal", "W/m2", "m/s", "m/s", "g/g", 
+        "kg/m2/s"))
+    
+    stepby <- list()  # list of time stepbys just in case they vary
+    
+    # Have to do 1 go first to initialize the dataframe
+    raw_train_data <- list()
+    tem <- ncdf4::nc_open(input_met[[1]])
+    dim <- tem$dim
+    for (j in seq_along(vars.info$CF.name)) {
+        if (exists(as.character(vars.info$CF.name[j]), tem$var) == FALSE) {
+            raw_train_data[[j]] <- NA
+        } else {
+            raw_train_data[[j]] <- ncdf4::ncvar_get(tem, as.character(vars.info$CF.name[j]))
+        }
+    }
+    names(raw_train_data) <- vars.info$CF.name
+    train_df <- data.frame(raw_train_data)
+    
+    # Figure out what temporal resolution the given data is in
+    # stepby[] helps us create the data sequence length
+    raw_train_data <- data.frame(raw_train_data)
+    if ((nrow(raw_train_data) == 17520) | (nrow(raw_train_data) == 17568)) {
+        stepby[1] <- 2
+    }
+    if ((nrow(raw_train_data) == 8760) | (nrow(raw_train_data) == 8784)) {
+        stepby[1] <- 1
+    }
+    if ((nrow(raw_train_data) == 1460) | (nrow(raw_train_data) == 1464)) {
+      stepby[1] <- (1/6)
+    }
+    # Add a time stamp
+    start_time <- as.POSIXlt(paste0(yr_seq[1], "-01-01"), tz = "UTC")
+    end_time <- as.POSIXlt(paste0(yr_seq[1] + 1, "-01-01"), tz = "UTC")
+    train_df$date <- seq.POSIXt(from = start_time, by = 60 * 60/stepby[[1]], 
+        length.out = nrow(train_df))
```

Comment:
@mdietze so I'm creating the `date` variable in the format that `lubridate` needs for the `aggregate` step. I also thought it might be useful to have a general date variable since not all of our netcdf's have the "days since xxxx" yet, but maybe I should assume they all do and it will force us to get it done. In the mean time, do you know of any documentation on `udunits` where I can transform `days since` to a `seq.POSIXt` string?